### PR TITLE
BUG: Specviz2d to load both s2d and x1d when both provided separately

### LINKED ISSUE #1711: Specviz2d does not load pair s2d/x1d
**Reporter: Brian Cherinka**

One of the original use cases of Specviz2d was to load a single associated pair of JWST s2d/x1d files.  This functionality no longer seems to work. When loading a pair of s2d/x1d files, the s2d files loads correctly but the x1d file does not.   See https://mast.stsci.edu/viz/ui/#/spectra?filename=jw02736-o008_s08883_nirspec_f170lp-g235m_s2d.fits&paired_file=jw02736-o008_s08883_nirspec_f170lp-g235m_x1d.fits

and the notebook code is
 
```python
s2d = 'jw02736-o008_s08883_nirspec_f170lp-g235m_s2d.fits'
x1d = 'jw02736-o008_s08883_nirspec_f170lp-g235m_x1d.fits'
s = Specviz2d()
s.load_data(spectrum_2d=s2d, spectrum_1d=x1d) 
```
 
[🐱](https://jira.stsci.edu/browse/JDAT-2834)

 --- 

 DISCLAIMER: This issue was autocreated by the Jdaviz Issue Creation Bot on behalf of the reporter. If any information is incorrect, please contact [Duy Nguyen](https://github.com/duytnguyendtn)

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to fix #1711 .

Screenshot 2022-10-10 205721

I don't know if we want to backport this or not. Feel free to change the milestone if we need to backport this.

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set?
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/specviz2d/helper.py [hunk 1]
```diff
@@ -195,7 +197,13 @@ def load_data(self, spectrum_2d=None, spectrum_1d=None, spectrum_1d_label=None,
                         color='warning', sender=self, timeout=10000)
                 self.app.hub.broadcast(msg)
 
+            else:
+                load_1d = True
+
         else:
+            load_1d = True
```
Comment (havok2063) [inline jdaviz/configs/specviz2d/helper.py:204]: Why do you set `load_1d = True` here  on the else condition for line 156, i.e. when `spectrum_2d` is provided? Everything works as expected.  Just trying to understand the flow here.

**Review by havok2063 — APPROVED** [review]
This PR works as expected.  I tried loading Specviz2d with an s2d/x1d pair, with only an s2d file, and with only a x1d file.

File: jdaviz/configs/specviz2d/helper.py (hunk 1 above)
Comment (pllim) [reply jdaviz/configs/specviz2d/helper.py:204]: This line only gets run when `spectrum_2d` is *not* provided (`None`). So I expect you will see the 1D loaded but nothing shown for the 2D viewer. Unless I have misunderstood the code?

Comment (havok2063) [reply jdaviz/configs/specviz2d/helper.py:204]: I meant the `load_1d` at line 204.  And yeah I think I see what you mean now.  It allows the use case when only the 1d is provided but not the 2d.  But would this cause an error at line 207, for the case when neither 2d nor 1d file is provided?  In that case, looks like `load_1d` gets set to true, the 1d spectrum tries to load but `spectrum_1d` is None?

Comment (pllim) [reply jdaviz/configs/specviz2d/helper.py:204]: But why would someone try to `load_data` without providing the data?

Comment (pllim) [reply jdaviz/configs/specviz2d/helper.py:207]: Good catch on the logic though. I added a check above. Does that look good to you?

Comment (havok2063) [reply jdaviz/configs/specviz2d/helper.py:207]: who knows **why** some people do the things they do.  I'm always of the opinion that someone somewhere will try the thing you think no one will try, and thus best safeguard against it.  Haha.   But this all looks good to me!

**Review by havok2063 — APPROVED** [review]

**Review by kecnry — APPROVED** [review]
Seems simple enough, thanks!  If you think this qualifies for the bugfix release, feel free to move it before merge (I don't see any reason to hold it back even if its not something that we make use of in official example notebooks).

**Comment by pllim** [comment]
@havok2063 , how soon do you need this fix? 😬

**Comment by havok2063** [comment]
@pllim I don't consider this a critical fix, so I'm fine with this getting merged in on whatever schedule y'all feel is right.   We still have to do testing for our update to 3.0.1, and aren't in a rush yet to push to prod.   If this gets in a 3.0.2 release, I can easily update our existing PR to bump to the new version.

**Comment by pllim** [comment]
Thanks, all! Okay, not backporting then. But I created a merge commit, so if we ever change our minds, it is not hard to backport.

**Pull request merged**